pyresample/resampler.py

```diff
@@ -121,6 +121,8 @@ def resample(self, data, cache_dir=None, mask_area=None, **kwargs):
         Returns (xarray.DataArray): Data resampled to the target area
 
         """
+        if self.source_geo_def == self.target_geo_def:
+            return data
```

Comment:
Do we want to do anything related to copying the data? For dask array stuff it would be a no-op since dask tasks are immutable, but for numpy it could mean a lot of extra memory usage in cases where the user was never going to make modifications anyway.

Otherwise this PR looks good to me.